### PR TITLE
chore(deps): upgrade react-router-dom to v6

### PR DESCRIPTION
**Pull request by AsToNlele**
# Description

Associated Jira ticket: [SPM-1858](https://issues.redhat.com/browse/SPM-1858)

How to test:
1. Make sure all the links redirect correctly
2. Approve
3. Profit?

# Checklist:

- [x] The commit message has the Jira ticket linked
- [x] PR has a short description
- [ ] Screenshots before and after the change are added
- [x] Tests for the changes have been added
- [ ] README.md is updated if necessary
- [ ] Needs additional dependent work

### CODE REVIEW

**Comment by AsToNlele** [comment]
I'll edit the description steps later

**Comment by AsToNlele** [comment]
After Frontend components have been upgraded, there is one less step :tada:

File: src/Routes.js
Comment (bastilian) [inline src/Routes.js:119]: Something is not right here. This should throw an error like "Router in Router not allowed".

Comment (bastilian) [reply src/Routes.js:119]: 🙈 my bad. I didn't see the `as` for the import. But maybe we can avoid that renaming. What if we use `AdvisorRoutes` for this const and not rename the `Routes` component from react router?

**Review by bastilian — APPROVED** [review]
Nice job! Thank you @AsToNlele!

**Comment by bastilian** [comment]
@AsToNlele Just out of precaution, we should get a second approval from @RedHatInsights/team-interact before merging.

**Review by bastilian — CHANGES_REQUESTED** [review]
Sorry, but I did find issues doublechecking on `/preview`.
There everything appears to fall apart. :/

File: src/Routes.js
Comment (bastilian) [inline src/Routes.js:88]: I think, this should use `chrome.appNavClick` like here[0] in order to work correctly. We have encountered with navigate where it does a full page refresh, which it shouldn't.

[0] https://github.com/RedHatInsights/compliance-frontend/pull/1892/files#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67R23

**Review by johnsonm325 — APPROVED** [review]
This seems to work okay.

There is a bug when you apply an os filter to the systems table and then try to refresh the page. But it exists in stage, so isn't related to this PR. I'm creating a card.

File: src/Routes.js
Comment (bastilian) [inline src/Routes.js:87]: According to the platform team, this whole part related to `APP_NAVIGATION` can and should be removed (see https://github.com/RedHatInsights/compliance-frontend/pull/1892/files#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67 for example)

**Pull request merged**
